core/src/main/java/jenkins/model/Jenkins.java

```diff
@@ -4722,6 +4723,7 @@ public static void _doScript(StaplerRequest req, StaplerResponse rsp, RequestDis
             }
 
             try {
+                ScriptListener.fireScriptEvent(req);
```

Comment:
why pass the complete request? Wouldn't it be enough to just pass the script itself? 
This code is also called when someone uses the script console of an agent. That information is not available from the request. That probably requires passing a reference to the computer instead of the Channel

Reply:
Good point. It would be nice to know on what machine the script was executed. I will have a look. Worst case will be a new / different parameter as you suggested.

And yes, I guess the request does not need to be passed.